internal/server/server.go

```diff
@@ -14,65 +15,85 @@ import (
 	newrelic "github.com/newrelic/go-agent"
 	"github.com/newrelic/go-agent/_integrations/nrgrpc"
 	"github.com/odpf/salt/log"
-	"github.com/odpf/salt/server"
+	"github.com/odpf/salt/mux"
 	"github.com/odpf/shield/internal/api"
 	"github.com/odpf/shield/internal/api/v1beta1"
 	"github.com/odpf/shield/internal/server/grpc_interceptors"
+	"github.com/odpf/shield/internal/server/health"
+	shieldv1beta1 "github.com/odpf/shield/proto/v1beta1"
 	"go.uber.org/zap"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/reflection"
 	"google.golang.org/grpc/status"
 )
 
-func registerHandler(ctx context.Context, s *server.MuxServer, gw *server.GRPCGateway, deps api.Deps) {
-	s.RegisterHandler("/admin*", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		fmt.Fprintf(w, "pong")
-	}))
-
-	s.RegisterHandler("/admin/ping", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		fmt.Fprintf(w, "pong")
-	}))
-
-	// grpc gateway api will have version endpoints
-	s.SetGateway("/admin", gw)
-	v1beta1.Register(ctx, s, gw, deps)
-}
-
 func Serve(
 	ctx context.Context,
 	logger log.Logger,
 	cfg Config,
 	nrApp newrelic.Application,
 	deps api.Deps,
-) (*server.MuxServer, error) {
-	s, err := server.NewMux(server.Config{
-		Port: cfg.Port,
-	}, server.WithMuxGRPCServerOptions(getGRPCMiddleware(cfg, logger, nrApp)))
+) error {
+	httpMux := http.NewServeMux()
+
+	grpcDialCtx, grpcDialCancel := context.WithTimeout(ctx, time.Second*5)
+	defer grpcDialCancel()
+
+	grpcConn, err := grpc.DialContext(
+		grpcDialCtx,
+		cfg.grpcAddr(),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithDefaultCallOptions(
+			grpc.MaxCallRecvMsgSize(cfg.GRPC.MaxRecvMsgSize),
+			grpc.MaxCallSendMsgSize(cfg.GRPC.MaxSendMsgSize),
+		))
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	gw, err := server.NewGateway("", cfg.Port, server.WithGatewayMuxOptions(
-		runtime.WithIncomingHeaderMatcher(customHeaderMatcherFunc(map[string]bool{cfg.IdentityProxyHeader: true}))),
+	grpcGateway := runtime.NewServeMux(
```

Comment:
CMIIW, but you need to add `runtime.WithIncomingHeaderMatcher(customHeaderMatcherFunc(map[string]bool{cfg.IdentityProxyHeader: true}))` here to retain the current behaviour.

Reply:
Oh my mistake, I accidentally removed this. thanks!